code/modules/mob/say.dm

```diff
@@ -101,6 +101,34 @@
 ///The amount of items we are looking for in the message
 #define MESSAGE_MODS_LENGTH 6
 
+/**
+ * Checks the inputted message for a custom say emote
+ * Basically it checks every message for "-"
+ * If a message contains it then it will mark everything that came before "-" as a custom say emote, IE: "stammers-", "cackles-", "screams-", "yells-", and everything after as the message
+ * If a message contains "-" but nothing after it then it will convert everything that came before "-" into an emote
+ * If a message doesn't contain "-" then it will simply return the input as a message
+ *
+ * Example
+ * * "mutters- hello" will be marked as a custom say emote of "mutters" and the message will be "hello"
+ * * and it will appear as Joe Average mutters, "hello"
+ * * "screams-" will be marked as a custom say emote of "screams" and it will appear as Joe Average screams
+ */
+/mob/proc/check_for_custom_say_emote(message, list/mods)
+	var/customsaypos = findtext(message, "-")
+	var/messagetextpos = 1
+	if(!customsaypos)
+		return message
+	if(findtext(message, " ", customsaypos + 1, customsaypos + 2))
+		messagetextpos = 2
+	if(is_banned_from(ckey, "Emote"))
+		return copytext(message, customsaypos + messagetextpos)
+	mods[MODE_CUSTOM_SAY_EMOTE] = lowertext(copytext_char(message, 1, customsaypos))
+	message = copytext(message, customsaypos + messagetextpos)
+	if(!message)
+		mods[MODE_CUSTOM_SAY_ERASE_INPUT] = TRUE
+		message = "an interesting thing to say"
```

Comment:
Are you sure it's a good idea to force people to say this?
Can you show any screenshots of how this looks?

Reply:
its a debug tool. if you see this the yame is broken.

Reply:
Yeah I noticed this is to track when someone says this, but I think you need to put a stacktrace when they say this.

Reply:
I agree that this should either CRASH if we are expecting a message to be passed in (and then !message has to be checked every time this proc is called) or we expect message to be nullable and just return empty string or nullable and accept the input.